packages/core/src/document.ts

```diff
@@ -580,39 +580,73 @@ class Document extends EventEmitter implements DocStateHolder {
     await this.dispatcher.publishTip(this.id, this.tip)
   }
 
+  /**
+   * Returns whether the given CID is in the log, and there is a later log entry that is an anchor commit
+   * @param cid
+   */
+  async _isCidAnchored(cid: CID): Promise<boolean> {
+    let foundCid = false
+    for (const entry of this.state.log) {
+      if (entry.cid == cid) {
+        foundCid = true
+      }
+
+      if (foundCid && entry.type == CommitType.ANCHOR) {
+        return true
+      }
+    }
+    return false
+  }
+
   /**
    * Request anchor for the latest document state
    */
   async anchor (): Promise<void> {
-    this._context.anchorService.on(this.id.toString(), async (asr: AnchorServiceResponse): Promise<void> => {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const doc = this
+    const tip: CID = this.tip
+
+    this._context.anchorService.on(this.id.toString(), async function listener (asr: AnchorServiceResponse) {
+      if (asr.cid.toString() != tip.toString()) { // todo don't use toString
```

Comment:
I have no idea why this is necessary, but I tried comparing the cids directly as well as comparing their `.bytes` and in both cases the CIDs were comparing unequal even when they were in fact the same.  Using `toString()` solved the issue, but it really shouldn't be necessary... Open to ideas on how to debug this further...